lib/dynamo/dynamo/jobs.py

```diff
@@ -20,31 +20,46 @@ def _get_job_count_for_month(user) -> int:
     return job_count_for_month
 
 
-def get_quota_status(user) -> Tuple[int, int, int]:
+def get_quota_status(user) -> Union[tuple[int, int, int], tuple[None, None, None]]:
     max_jobs = get_max_jobs_per_month(user)
-    previous_jobs = _get_job_count_for_month(user)
-    remaining_jobs = max(max_jobs - previous_jobs, 0)
+
+    if max_jobs is not None:
+        previous_jobs = _get_job_count_for_month(user)
+        remaining_jobs = max(max_jobs - previous_jobs, 0)
+    else:
+        previous_jobs = None
+        remaining_jobs = None
+
     return max_jobs, previous_jobs, remaining_jobs
 
 
 def put_jobs(user_id: str, jobs: List[dict], fail_when_over_quota=True) -> List[dict]:
     table = DYNAMODB_RESOURCE.Table(environ['JOBS_TABLE_NAME'])
     request_time = format_time(datetime.now(timezone.utc))
 
+    priority_override = get_priority(user_id)
     max_jobs, previous_jobs, remaining_jobs = get_quota_status(user_id)
-    if len(jobs) > remaining_jobs:
+
+    if max_jobs is not None and len(jobs) > remaining_jobs:
         if fail_when_over_quota:
             raise QuotaError(f'Your monthly quota is {max_jobs} jobs. You have {remaining_jobs} jobs remaining.')
         jobs = jobs[:remaining_jobs]
 
+    if priority_override is not None:
+        priority = lambda _: priority_override  # noqa: E731
+    elif max_jobs is None:
+        priority = lambda _: 0  # noqa: E731
+    else:
+        priority = lambda job_index: max(9999 - previous_jobs - job_index, 0)  # noqa: E731
```

Comment:
I like the approach, but I'm curious if there's a way to move this chunk of logic into a separate function to hide the details?

Reply:
Also wondering whether the if/elif/else should specifically look for `previous_jobs is not None`, since that's the value being referenced in the `else` block?

Reply:
Here's an example of using a separate function and checking `previous_jobs` rather than `max_jobs`:

```python3
def _get_job_priority(priority_override: Optional[int], previous_jobs: Optional[int]):
    if priority_override is not None:
        priority = lambda _: priority_override
    elif previous_jobs is None:
        priority = lambda _: 0
    else:
        priority = lambda job_index: max(9999 - previous_jobs - job_index, 0)
    return priority
```
It could of course be refactored to instead check `previous_jobs is not None`. But regardless, I feel like this approach makes it more difficult to understand that we default to priority `0` when the user has no quota. I think checking `max_jobs is None` makes it more clear; but it seems a bit silly to pass `max_jobs` into the helper function just to check if it's `None`. Perhaps it would make sense to set a boolean somewhere, maybe in the `tuple` returned by `get_quota_status`, named something like `no_quota` or `unlimited_jobs`, to be the single source of truth for whether the user has a quota, since right now there are three variables (`max_jobs, previous_jobs, remaining_jobs`) that are all `None` when the user has no quota, and all `not None` otherwise, so it's not obvious which one of those three should be checked to determine if the user has a quota.